mapper/orm/src/main/java/org/hibernate/search/mapper/orm/loading/impl/HibernateOrmNonEntityIdPropertyEntityLoadingStrategy.java

```diff
@@ -144,4 +144,11 @@ private AssertionFailure multipleTypesException(Set<? extends LoadingTypeContext
 								.collect( Collectors.toList() )
 		);
 	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	protected TypeQueryFactory<E, I> createFactory(EntityMappingType commonSuperType) {
+		return (TypeQueryFactory<E, I>) TypeQueryFactory.create( sessionFactory, commonSuperType,
+				documentIdSourcePropertyName );
+	}
```

Comment:
The fact that we have to implement this method differently in the entity-id/non-entity-id subclasses makes me think that some conditions in the constructor of `HqlTypeQueryFactory` are probably unnecessary... we could have two separate constructors / factory methods.